coffee/base.py

```diff
@@ -445,6 +445,9 @@ def __init__(self, symbol, rank=(), offset=()):
     def operands(self):
         return [self.symbol, self.rank, self.offset], {}
 
+    def reconstruct(self):
+        return Symbol(self.symbol, self.rank, self.offset)
```

Comment:
`reconstruct` should take new arguments to reconstruct with.  The default reconstruct method does this.  Why do you need this change (which causes tests to fail)?